src/services/ui/src/pages/actions/index.tsx

```diff
@@ -5,22 +5,83 @@ import { WithdrawPackage } from "@/pages/actions/WithdrawPackage";
 import { RespondToRai } from "@/pages/actions/RespondToRai";
 import { Action } from "shared-types";
 import { WithdrawRai } from "./WithdrawRai";
+import { useGetItem, useGetPackageActions } from "@/api";
+import {
+  Alert,
+  BreadCrumbs,
+  LoadingSpinner,
+  SimplePageContainer,
+} from "@/components";
+import { ModalProvider } from "@/pages/form/modals";
+import { detailsAndActionsCrumbs } from "@/pages/actions/actions-breadcrumbs";
 
-export const ActionFormIndex = () => {
-  const { type } = useParams("/action/:id/:type");
+const ActionFormSwitch = () => {
+  const { id, type } = useParams("/action/:id/:type");
+  const {
+    data: item,
+    isLoading: itemIsLoading,
+    error: itemError,
+  } = useGetItem(id!);
+  const {
+    data: actions,
+    isLoading: actionsAreLoading,
+    error: actionsError,
+  } = useGetPackageActions(id!);
+  // Safety bolt-on to limit non-null assertions needed
+  if (!id || !type) return <Navigate path="/" />;
+
+  // Non-form renders
+  if (itemIsLoading || actionsAreLoading) return <LoadingSpinner />;
+  if (itemError)
+    return (
+      <Alert className="my-2 max-w-2xl" variant="destructive">
+        <strong>ERROR getting item: </strong>
+        {itemError.response.data.message}
+      </Alert>
+    );
+  if (actionsError)
+    return (
+      <Alert className="my-2 max-w-2xl" variant="destructive">
+        <strong>ERROR getting available actions: </strong>
+        {actionsError.response.data.message}
+      </Alert>
+    );
+  if (!actionsError && !actions?.actions.includes(type))
+    return (
+      <Alert className="my-2 max-w-2xl" variant="destructive">
+        <strong>ERROR, invalid action: </strong>
+        You cannot perform {type} on this package.
+      </Alert>
+    );
+
+  // Form renders
   switch (type) {
     case Action.WITHDRAW_PACKAGE:
-      return <WithdrawPackage />;
+      return <WithdrawPackage item={item} />;
     case Action.ENABLE_RAI_WITHDRAW:
     case Action.DISABLE_RAI_WITHDRAW:
-      return <ToggleRaiResponseWithdraw />;
+      return <ToggleRaiResponseWithdraw item={item} />;
     case Action.ISSUE_RAI:
-      return <RaiIssue />;
+      return <RaiIssue item={item} />;
     case Action.WITHDRAW_RAI:
-      return <WithdrawRai />;
+      return <WithdrawRai item={item} />;
     case Action.RESPOND_TO_RAI:
-      return <RespondToRai />;
+      return <RespondToRai item={item} />;
     default:
       return <Navigate path="/" />;
   }
 };
+
+export const ActionFormIndex = () => {
+  const { id, type } = useParams("/action/:id/:type");
+  return (
+    <SimplePageContainer>
+      <ModalProvider>
```

Comment:
Notably we _add_ a dependency, but isolate its usage to a point that wraps all action forms instead of wrapping them individually, and provides a replicable solution for future form indexes (i.e. Submission forms when refactored to a dynamic-path approach like Actions forms).